include/gz/sim/components/Environment.hh

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GZ_SIM_ENVIRONMENT_HH_
+#define GZ_SIM_ENVIRONMENT_HH_
+
+#include <memory>
+#include <string>
+
+#include <gz/common/DataFrame.hh>
+#include <gz/math/SphericalCoordinates.hh>
+#include <gz/math/TimeVaryingVolumetricGrid.hh>
+
+#include <gz/sim/components/Factory.hh>
+#include <gz/sim/components/Component.hh>
+
+namespace gz
+{
+namespace sim
+{
+// Inline bracket to help doxygen filtering.
+inline namespace GZ_SIM_VERSION_NAMESPACE {
+namespace components
+{
+  /// \brief Environment data across time and space. This is useful to
+  /// introduce physical quantities that may be of interest even if not
+  /// modelled in simulation. 
+  struct EnvironmentalData
+  {
+    using T = math::InMemoryTimeVaryingVolumetricGrid<double>;
+    using FrameT = common::DataFrame<std::string, T>;
+    using ReferenceT = math::SphericalCoordinates::CoordinateType;
+
+    /// \brief Instantiate environmental data.
```

Comment:
Knit: There's a bit of whitespace at EOL issue here. Should be caught when running `make codecheck`

Reply:
Done in https://github.com/gazebosim/gz-sim/pull/1616/commits/36c444dd354fa540784d1dbdd4e7fbda19ced8eb.